pkg/share/manager/json/json.go

```diff
@@ -393,7 +398,29 @@ func (m *mgr) ListReceivedShares(ctx context.Context, filters []*collaboration.F
 			rss = append(rss, rs)
 		}
 	}
-	return rss, nil
+
+	// if there is a mix-up of shares of type group and shares of type user we need to deduplicate them, since it points
+	// to the same resource. Leave the more explicit and hide the more explicit. In this case we hide the group shares
+	// and return the user share to the user.
+	filtered := make([]*collaboration.ReceivedShare, 0)
+	filtered = append(filtered, rss...)
+
+	for i := range rss {
+		for j := range rss {
+			if rss[i].Share.ResourceId.GetOpaqueId() == rss[j].Share.ResourceId.GetOpaqueId() {
+				if rss[i].Share.GetGrantee().GetType() == provider.GranteeType_GRANTEE_TYPE_GROUP && rss[j].Share.GetGrantee().GetType() == provider.GranteeType_GRANTEE_TYPE_USER {
+					if rss[i].State == rss[j].State {
```

Comment:
Not sure about this. If the permissions differ for the shares, shouldn't be return both?